### PR TITLE
Refactor client/streams to support upgrade and multisend using chans

### LINKED ISSUE #6: upgrade connection to stream

### PR DESCRIPTION
**Pull request by taubyte0**

### CODE REVIEW

File: streams/client/client.go [hunk 1]
```diff
@@ -234,64 +263,87 @@ func (c *Client) connect(peer peerCore.AddrInfo) (network.Stream, bool, error) {
...
+	if timeout == 0 {
+		timeout = SendToPeerTimeout
+	}
+
+	if minStreams > MaxStreamsPerSend {
+		return nil, fmt.Errorf("threashold %d exceeds MaxStreamsPerSend", minStreams)
+	}
+
+	ctx, ctxC := context.WithTimeout(c.ctx, timeout)
+	cmdDD, _ := ctx.Deadline()
```
Comment (tafseer-khan) [inline streams/client/client.go:327]: move this down, not using this till later

File: streams/client/client.go [hunk 2]
```diff
@@ -30,12 +30,60 @@ type Client struct {
...
 }
 
-type response struct {
-	peerCore.ID
+type Request struct {
+	client     *Client
+	to         []peerCore.ID
+	cmd        string
+	body       command.Body
+	cmdTimeout time.Duration
```
Comment (tafseer-khan) [inline streams/client/client.go:38]: Why not ctx instead of timeout

File: streams/client/client.go [hunk 3]
```diff
@@ -50,6 +98,8 @@ var (
 	EstablishStreamTimeout time.Duration = 5 * time.Second
 	SendToPeerTimeout      time.Duration = 10 * time.Second
```
Comment (tafseer-khan) [inline streams/client/client.go:99]: Shoud these be exported? Considering we now set these in request, and this is default if not set

File: streams/client/client.go [hunk 4]
```diff
@@ -61,101 +111,75 @@ func (c *Client) Context() context.Context {
...
-
-	strm, err := c.node.Peer().NewStream(ctx, pid, protocol.ID(c.path))
-	if err != nil {
-		return nil, fmt.Errorf("peer new stream failed with: %w", err)
-	}
-	defer strm.Reset()
-
-	if err = strm.SetWriteDeadline(now.Add(SendTimeout)); err != nil {
-		return nil, fmt.Errorf("set write deadline failed with: %w", err)
+func (c *Client) New(cmd string, opts ...Option) *Request {
```
Comment (tafseer-khan) [inline streams/client/client.go:125]: My opinion c.Request(...) sounds better... but either way is fine

**Review by tafseer-khan — CHANGES_REQUESTED** [review]

File: streams/client/client.go (hunk 2 above)
Comment (taubyte0) [reply streams/client/client.go:38]: coz network read/write are blocking using time

File: streams/client/client.go (hunk 3 above)
Comment (taubyte0) [reply streams/client/client.go:99]: yes. to alter default

File: streams/client/client.go (hunk 4 above)
Comment (taubyte0) [reply streams/client/client.go:125]: New like net/http

File: streams/client/client.go (hunk 1 above)
Comment (taubyte0) [reply streams/client/client.go:327]: easy to relate to ctx here

**Pull request merged**